pkg/storage/ec/client.go

```diff
@@ -126,6 +126,16 @@ func (ec *ecClient) Put(ctx context.Context, nodes []*pb.Node, rs eestream.Redun
 		}
 	}
 
+	/* clean up the partially uploaded segment's pieces */
+	defer func() {
+		select {
+		case <-ctx.Done():
+			ctx = context.Background()
```

Comment:
Here we are replacing the `ctx` variable with `context.Background()`. This may have some undesired side-effect, having in mind that `defer mon.Task()(&ctx)(&err)` is yet to be executed.

It would be safer if we remove this line and modify delete call to:

```go
ec.Delete(ctx.Background(), nodes, pieceID)
```

Reply:
I really like the use of nodes...than successful nodes...